x-pack/plugins/reporting/server/plugin.ts

```diff
@@ -15,47 +14,56 @@ import { setFieldFormats } from './services';
 import { ReportingSetup, ReportingSetupDeps, ReportingStart, ReportingStartDeps } from './types';
 import { registerReportingUsageCollector } from './usage';
 
+declare module 'src/core/server' {
+  interface RequestHandlerContext {
+    reporting?: ReportingCore | null;
```

Comment:
This allows `'reporting'` to be the first parameter to `registerRouteHandlerContext` as the string must be a key of `RequestHandlerContext`